app/api/dao/program.py

```diff
@@ -0,0 +1,248 @@
+import ast
+import logging
+from http import HTTPStatus
+from app.database.models.bit_schema.organization import OrganizationModel
+from app.database.models.bit_schema.program import ProgramModel
+from app.database.models.bit_schema.user_extension import UserExtensionModel
+from app import messages
+from app.api.request_api_utils import AUTH_COOKIE, get_request, http_response_checker
+from app.utils.bitschema_utils import *
+from app.utils.bit_constants import MAX_PROGRAMS_PER_PAGE
+
+
+class ProgramDAO:
+    
+    """Data Access Object for Program functionalities"""
+
+    @staticmethod
+    def get_program_by_program_id(organization_id, program_id, token):
+        """Retrieves a program which ID is passed as parameter.
+
+        Arguments:
+            organization_id: The ID of the organization where the program is offered.
+            program_id: The ID of the program.
+            token: The access token retrieved when login. 
+            
+        Returns:
+            The ProgramModel class represented by its ID was searched.
+        """
+        
+        organization = OrganizationModel.find_by_id(organization_id)
+        if not organization:
+            return messages.NO_ORGANIZATION_FOUND, HTTPStatus.NOT_FOUND
+        user = http_response_checker(get_request("/users/"+f"{organization.rep_id}", token, params=None))
+        if not user.status_code == 200:
+            return user.message, user.status_code
+        try:
+            program = ProgramModel.find_by_id(program_id)
+            if program.organization_id != organization_id:
+                return messages.PROGRAM_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND
+        except AttributeError:
+            return messages.PROGRAM_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND
+        return get_program(program, organization, user.message)
+
+        
+    @staticmethod
+    def get_all_programs_by_organization(organization_id, token):
+
+        """Retrieves all the programs that are offered by the organization which ID is passed as parameter.
+
+        Arguments:
+            organization_id: The ID of the organization where the program is offered.
+            token: The access token retrieved when login. 
+            
+        Returns:
+            The list of Programs offered by the orgranization which ID was searched.
+        """
+
+        organization = OrganizationModel.find_by_id(organization_id)
+        if not organization:
+            return messages.NO_ORGANIZATION_FOUND, HTTPStatus.NOT_FOUND
+        user = http_response_checker(get_request("/users/"+f"{organization.rep_id}", token, params=None))
+        if not user.status_code == 200:
+            return user.message, user.status_code
+
+        programs = ProgramModel.get_all_programs_by_organization(organization_id)
+        if not programs:
+            return messages.NO_PROGRAM_FOUND, HTTPStatus.NOT_FOUND
+        list_of_programs = []
```

Comment:
Lines 68-70 (even 71) can be condensed into 1 line using list comprehension. Could you please do that?

Reply:
Done, @ramitsawhney27 . It surely looks better now. Thanks heaps for the tips. 😉